documentation/Advanced Features/Objects and Extensions.md

```diff
@@ -0,0 +1,269 @@
+# Objects and Extension Methods
+
+Vyxal is already turing complete, and can be used to easily solve many problems.
+However, say there was a situation where you needed to model a complex real-world
+object, such as a car. You could use two lists to map keys to values, but that
+would require making sure that the two lists are always in sync. Objects take
+away this hassle, and allow you to model real-world objects in a way that is
+convienient and easy to use.
+
+## The OOP Model
+
+Unlike most OOP languages, Vyxal does not have traditional classes. Instead, it
+has struct-like objects. These objects contain only data, and no methods. To add
+methods to an object, you can use extension methods (covered later).
+
+There is also no inheritance, meaning that composition is the only way to reuse
+code. This is not a bad thing, as it makes the implementation of objects much
+simpler.
+
+## Defining Objects
+
+To define an object:
+
+```
+#:O ObjectName | #$restrictedMember #=privateMember <data> #$restrictedMemberWithValue
+<data> #=privateMemberWithValue #!publicMember}

```

Comment:
```suggestion
#:O ObjectName |
  #$restrictedMember
  #=privateMember
  <data> #$restrictedMemberWithValue
  <data> #=privateMemberWithValue #!publicMember}
```
Not a big deal (you don't have to accept this) but it might be clearer (I was confused at first because I thought the first `<data>` belonged to the first `#=privateMember` before realizing this is a stack-based language)